_posts/2015-09-xx-creating-systems-with-pipelining.md

```diff
@@ -0,0 +1,52 @@
+---
+layout: post
+title: "creating systems with pipelining"
+date: "2015-09-xx"
+icon: wrench
+tags: pipelines, virtual mashines, development systems
+categories: Application Management
+authors: ["Carsten", "Karsten"]
+---
+
+Today I will introduce some special topic to our readers, called [pipelining](https://en.wikipedia.org/wiki/Pipeline_(software)). When we at ePages speak about pipelining, we understand it as an automatic workflow, integrating many steps using different tools to reach a specific goal. The goal we had was to create a simple way to provide a highly specialized system. This system can be for development, external or even live.
+The setup used here is a combination of different software and tools. One of them is [vmware](http://www.vmware.com/) with [vCenter](https://www.vmware.com/products/vcenter-server) and some [ESXi](https://www.vmware.com/products/vsphere/features/esxi-hypervisor) servers to offer and manage systems in a virtual environment.
+Another tool we use is called [i-doit](http://www.i-doit.org/) and is used for creation and documentation of inventories and network related information.
+As you may read in a previous [post](https://developer.epages.com/blog/2015/06/25/infrastructure-as-code.html) we often use [Jenkins](https://jenkins-ci.org/) for [Continuous Integration](http://www.martinfowler.com/articles/continuousIntegration.html), so it's one of our favourite tools to automate processes.
+The key of an automated OS install is the combination of [iPXE](http://ipxe.org/) and kickstart isos, for example preseeds for [Debian](https://wiki.debian.org/DebianInstaller/Preseed) or [Centos/RHEL](http://www.centos.org/docs/4/html/rhel-sag-en-4/s1-kickstart2-file.html).
+Last but not least when it comes to system specialization [Linux](https://www.linux.com/) itself should be mentioned :P.
+To speak with the different interfaces, programming languages like [`Perl`](https://www.perl.org/) and [`Python`](https://www.python.org/) or scripting languages like [`PHP`](https://secure.php.net/) and [`bash`](http://www.gnu.org/software/bash/) are used.
+
+{% image blog-pipeline-tools.png %} {% endimage %}
+
+# The pipeline
+
+The basic idea is simple. First off we need a system.
+
+## The system
+
+This is fairly simple, due to the [vSphere Cloud Plugin](https://wiki.jenkins-ci.org/display/JENKINS/vSphere+Cloud+Plugin) for Jenkins, it is easy to construct a job building a basic virtual machine from a template.
+These templates are predefined with a set of CPU,RAM and Disc space and an OS specific mounted iPXE kickstart iso file. If it's needed, you can create and run an additional job changing the values of the machine before the install starts.
```

Comment:
Change to something like that: These templates are predefined with a set of CPU, RAM, disc space as well as an iPXE kickstart iso file that is specifically OS mounted.
("OS specific mounted iPXE kickstart iso file" is really hard to read)